lib/demo_web/controllers/user_auth.ex

```diff
@@ -3,111 +3,85 @@ defmodule DemoWeb.UserAuth do
-  if you are not using LiveView.
   """
   def login_user(conn, user, params \\ %{}) do
     token = Accounts.generate_user_session_token(user)
-    user_return_to = get_session(conn, :user_return_to)
 
-    conn
-    |> renew_session()
-    |> put_session(:user_token, token)
-    |> put_session(:live_socket_id, "users_sessions:#{Base.url_encode64(token)}")
-    |> maybe_write_remember_me_cookie(token, params)
-    |> redirect(to: user_return_to || signed_in_path(conn))
-  end
+    conn =
+      conn
+      |> write_remember_me_cookie(token, params)
 
-  defp maybe_write_remember_me_cookie(conn, token, %{"remember_me" => "true"}) do
-    put_resp_cookie(conn, @remember_me_cookie, token, @remember_me_options)
+    {conn, token}
   end
 
-  defp maybe_write_remember_me_cookie(conn, _token, _params) do
-    conn
+  defp write_remember_me_cookie(conn, token, %{"remember_me" => "true"}) do
+    remember_me_options_with_max_age = Keyword.put(@remember_me_options, :max_age, @max_age)
+    put_resp_cookie(conn, @remember_me_cookie, token, remember_me_options_with_max_age)
   end
 
-  # This function renews the session ID and erases the whole
-  # session to avoid fixation attacks. If there is any data
-  # in the session you may want to preserve after login/logout,
-  # you must explicitly fetch the session data before clearing
-  # and then immediately set it after clearing, for example:
-  #
-  #     defp renew_session(conn) do
-  #       preferred_locale = get_session(conn, :preferred_locale)
-  #
-  #       conn
-  #       |> configure_session(renew: true)
-  #       |> clear_session()
-  #       |> put_session(:preferred_locale, preferred_locale)
-  #     end
-  #
-  defp renew_session(conn) do
-    conn
-    |> configure_session(renew: true)
-    |> clear_session()
+  defp write_remember_me_cookie(conn, token, _params) do
+    put_resp_cookie(conn, @remember_me_cookie, token, @remember_me_options)
   end
 
   @doc """
   Logs the user out.
-
-  It clears all session data for safety. See renew_session.
   """
   def logout_user(conn) do
-    user_token = get_session(conn, :user_token)
+    user_token = conn.private[:user_token]
     user_token && Accounts.delete_session_token(user_token)
 
-    if live_socket_id = get_session(conn, :live_socket_id) do
-      DemoWeb.Endpoint.broadcast(live_socket_id, "disconnect", %{})
-    end
-
     conn
-    |> renew_session()
+    |> put_private(:user_token, nil)
     |> delete_resp_cookie(@remember_me_cookie)
-    |> redirect(to: "/")
   end
 
   @doc """
-  Authenticates the user by looking into the session
+  Authenticates the user by looking into the header
   and remember me token.
   """
   def fetch_current_user(conn, _opts) do
     {user_token, conn} = ensure_user_token(conn)
     user = user_token && Accounts.get_user_by_session_token(user_token)
-    assign(conn, :current_user, user)
+
+    conn
+    |> assign(:current_user, user)
+    |> put_private(:user_token, user_token)
```

Comment:
Is it ok to set the token to the send value even though it might not be a valid token? Because in case the `user` is nil, I was thinking we set the`user_token` to nil too.